marathon_quiz.cpp

```diff
@@ -191,3 +191,117 @@ void freeQuestions(struct Question* head) {
         free(temp);
     }
 }
+
+//Functions for the main process for player creation
+
+//Node creation
+Player* createPlayer(int id, char *nickname, float score){
+    Player *newPlayer=(Player*)malloc(sizeof(Player));
+    newPlayer->id=id;
+    newPlayer->maxScore=score;
+    strcpy(newPlayer->nickname,nickname);
+    newPlayer->next=NULL;
+    newPlayer->prev=NULL;
+    return newPlayer;
+}
+
+//Insert new player in order from the highest score to the lowest
+void insertSortedPlayer(Player **head, Player *newPlayer, bool &inserted){
+    //Check for repeated nicknames
+    Player *current=*head;
+    if(current!=NULL){
```

Comment:
Please refacto this code. I think a better approach is create another function to return is a player is already in the list.

Reply:
I agree with this

Reply:
Changes have been made

Reply:
Awesome !!